client/qiskit_serverless/core/files.py

```diff
@@ -179,14 +179,43 @@ def provider_list(self, function: QiskitFunction) -> List[str]:
             )
         return response_data.get("results", [])
 
-    def delete(self, file: str, provider: Optional[str] = None) -> Optional[str]:
+    def delete(
+        self, file: str, function: QiskitFunction, provider: Optional[str] = None
+    ) -> Optional[str]:
         """Deletes file uploaded or produced by the programs,"""
         tracer = trace.get_tracer("client.tracer")
         with tracer.start_as_current_span("files.delete"):
             response_data = safe_json_request_as_dict(
                 request=lambda: requests.delete(
                     os.path.join(self._files_url, "delete"),
-                    data={"file": file, "provider": provider},
+                    data={
```

Comment:
I think this `data` field needs to be renamed to `params` now. Correct me if I'm wrong.